stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authz-filter.bal

```diff
@@ -0,0 +1,134 @@
+    function (http:Response response, http:FilterContext context) returns (http:FilterResult) filterResponse;
+}
+
+@Description {value:"Initializer for AuthzFilter"}
+public function <AuthzFilter filter> AuthzFilter () {
+    filter.filterRequest = authzRequestFilterFunc;
+    filter.filterResponse = responseFilterFunc;
+}
+
+@Description {value:"Initializes the AuthzFilter"}
+public function <AuthzFilter filter> init () {
+    authzHandlerChain = createAuthzHandlerChain();
+}
+
+@Description {value:"Stops the AuthzFilter"}
+public function <AuthzFilter filter> terminate () {
+}
+
+@Description {value:"Filter function implementation which tries to authorize the request"}
+@Param {value:"request: Request instance"}
+@Param {value:"context: FilterContext instance"}
+@Return {value:"FilterResult: Authorization result to indicate if the request can proceed or not"}
+public function authzRequestFilterFunc (http:Request request, http:FilterContext context) returns (http:FilterResult) {
+    // check if this resource is protected
+    string scope = getScopeForResource(context);
+    boolean authorized;
+    if (scope != "") {
+        authorized = authzHandlerChain.handle(request, scope, context.resourceName);
+    } else {
+        // if scopes are not defined, no need to authorize
+        return createAuthzResult(true);
+    }
+    return createAuthzResult(authorized);
+}
+
+@Description {value:"Creates an instance of FilterResult"}
+@Param {value:"authorized: authorization status for the request"}
+@Return {value:"FilterResult: Authorization result to indicate if the request can proceed or not"}
+function createAuthzResult (boolean authorized) returns (http:FilterResult) {
+    http:FilterResult requestFilterResult;
+    if (authorized) {
+        requestFilterResult = {canProceed:true, statusCode:200, message:"Successfully authorized"};
+    } else {
+        requestFilterResult = {canProceed:false, statusCode:403, message:"Authorization failure"};
+    }
+    return requestFilterResult;
+}
+
+@Description {value:"Retrieves the scope for the resource, if any"}
+@Param {value:"context: FilterContext object"}
+@Return {value:"string: Scope name if defined, else null"}
+function getScopeForResource (http:FilterContext context) returns (string) {
+    string scope = getAuthzAnnotation(internal:getResourceAnnotations(context.serviceType,
+                                                                      context.resourceName));
+    match scope {
+        string scopeVal => {
+            return scopeVal;
+        }
+        error err => {
+            // if not found in resource level, check in service level
+            scope = getAuthzAnnotation(internal:getServiceAnnotations(context.serviceType));
+            match scope {
+                string scopeVal => {
+                    return scopeVal;
+                }
+                error err => {
+                    // if the scope is still null, means authorization is not needed.
```

Comment:
Isn't this a dangerous assumption? If there is an error, you are saing authz is not needed.

Reply:
changed the downstream method to return a null if no scopes are found and used a match statement.